packages/react-core/src/components/ToggleGroup/examples/ToggleGroupCompact.tsx

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import { ToggleGroup, ToggleGroupItem } from '@patternfly/react-core';
+
+export const ToggleGroupCompact: React.FunctionComponent = () => {
+  const [isSelected, setIsSelected] = React.useState({
+    'toggle-group-compact-1': false,
+    'toggle-group-compact-2': false
+  });
+  const handleItemClick = (isSelected: boolean, event: React.MouseEvent<any> | React.KeyboardEvent | MouseEvent) => {
+    const id = event.currentTarget.id;
+    setIsSelected(prevIsSelected => {
+      prevIsSelected[id] = isSelected;
+      return { ...prevIsSelected };
+    });
```

Comment:
An alternative to what you have here + for the other examples could be:

```suggestion
    setIsSelected(prevIsSelected => {
      return { ...prevIsSelected, [id]: isSelected };
    });
```

Just to avoid mutating the previous state object.